From 68f7a3035da23bb74ee9369340fa6506710c43e3 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 26 Aug 2014 12:49:36 -0400
Subject: [tools] cpupower: cpupower info -b should return 0 on success, not the perf bias value

Message-id: <1409057382-26432-26-git-send-email-prarit@redhat.com>
Patchwork-id: 89093
O-Subject: [RHEL7.1 PATCH BZ 1133893 25/31] cpupower: cpupower info -b should return 0 on success, not the perf bias value
Bugzilla: 1133893
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1133893

commit 3fc5a0e51aef4503b6a06ef35409370eed568684
Author: Thomas Renninger <trenn@suse.de>
Date:   Tue May 13 12:41:44 2014 +0200

    cpupower: cpupower info -b should return 0 on success, not the perf bias value

    Signed-off-by: Thomas Renninger <trenn@suse.de>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/cpupower/utils/cpupower-info.c b/tools/power/cpupower/utils/cpupower-info.c
index 3f68632..0ac2577 100644
--- a/tools/power/cpupower/utils/cpupower-info.c
+++ b/tools/power/cpupower/utils/cpupower-info.c
@@ -125,11 +125,12 @@ int cmd_info(int argc, char **argv)
   if (params.perf_bias) {
    ret = msr_intel_get_perf_bias(cpu);
    if (ret < 0) {
-    printf(_("Could not read perf-bias value\n"));
-    break;
+    fprintf(stderr,
+   _("Could not read perf-bias value[%d]\n"), ret);
+    exit(EXIT_FAILURE);
    } else
     printf(_("perf-bias: %d\n"), ret);
   }
  }
- return ret;
+ return 0;
 }
-- 
1.7.1